app/main.py

```
@@ -1,34 +1,137 @@
+        for (row, col), ship in self.field.items():
+            for deck in ship.decks:
+                if deck.row == row and deck.column == col:
+                    if ship.is_drowned:
+                        grid[row][col] = "x"
+                    elif not deck.is_alive:
+                        grid[row][col] = "*"
+                    else:
+                        grid[row][col] = u"\u25A1"
+        for row in grid:
+            print("\t".join(row))
+
+    def _validate_field(self) -> None:
+        if len(self.ships) != 10:
+            raise ValueError("Ships must be exactly 10.")
+
+        ship_sizes = defaultdict(int)
+        occupied = set()
+
+        for ship in self.ships:
+            ship_sizes[len(ship.decks)] += 1
+
+            for deck in ship.decks:
+                for direction_row in [-1, 0, 1]:
+                    for direction_column in [-1, 0, 1]:
+                        near_row, near_colum = \
+                            (
+                                deck.row + direction_row,
+                                deck.column + direction_column
+                            )
+                        if (
+                            (near_row, near_colum) in occupied
+                            and (near_row, near_colum) not in
+                            [
+                                (temp_deck.row, temp_deck.column)
+                                for temp_deck in ship.decks
+                            ]
+                        ):
+                            raise ValueError(
+                                f"Ships are too close to each other at: "
+                                f"({near_row}, {near_colum})."
+                            )
+            for deck in ship.decks:
+                occupied.add((deck.row, deck.column))
+
+        if (
+            ship_sizes[1] != 4
+            and ship_sizes[2] != 3
+            and ship_sizes[3] != 2
+            and ship_sizes[4] != 1
```

Issue: The validation logic for ship sizes uses 'and' instead of 'or'. According to the checklist, you must ensure there are exactly 4 ships of size 1, 3 of size 2, 2 of size 3, and 1 of size 4. The current logic only raises an error if *all* counts are incorrect, but it should raise an error if *any* of the counts are incorrect. Replace 'and' with 'or' in this condition.